backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Application.java

```diff
@@ -176,7 +176,11 @@ protected void loop () {
 					window.makeCurrent();
 					currentWindow = window;
 				}
-				if (targetFramerate == -2) targetFramerate = window.getConfig().foregroundFPS;
+				if (targetFramerate == -2 && window.isFocused()) targetFramerate = window.getConfig().foregroundFPS;
+				synchronized (lifecycleListeners) {
+					haveWindowsRendered |= window.update();
+				}
+				if (targetFramerate == -2 && !window.isFocused()) targetFramerate = window.getConfig().backgroundFPS;
```

Comment:
This looks odd to me; why is this a separate if statement, where it is after the window updates, instead of an else block that runs when the window isn't focused?